core/src/Platforms/iOS/TokenCacheAccessor.cs

```diff
@@ -284,6 +284,13 @@ private SecStatusCode Save(string account, string service, string generic, int t
                 secStatusCode = SecKeyChain.Add(recordToSave);
             }
 
+            if(secStatusCode == SecStatusCode.MissingEntitlement)
+            {
+                throw CoreExceptionFactory.Instance.GetClientException(
```

Comment:
It would be nice if the resulting exception message contained the keychain group that the accessor is attempting to save to, like:

`"The application does not have the keychain access group '{0}' enabled in the Entitlements.plist."`

Reply:
Makes sense to also add it here. The developer has specified it in their own code though, thus they should know?

---
In reply to: [224734206](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1318#discussion_r224734206) [](ancestors = 224734206)

Reply:
@henrik-me Not necessarily - the TokenCacheAccessor might be using its default value for the keychain group, in which case it's super helpful to know what it's trying to do.

Reply:
Haha, I think I replied before you edited your comment. :)